pkg/storage/snapshot/restore.go

```diff
@@ -261,8 +271,13 @@ func (ctrl *VMRestoreController) handleVMRestoreDeletion(vmRestore *snapshotv1.V
 
 	updateRestoreCondition(vmRestoreCpy, newProgressingCondition(corev1.ConditionFalse, "VM restore is deleting"))
 	updateRestoreCondition(vmRestoreCpy, newReadyCondition(corev1.ConditionFalse, "VM restore is deleting"))
+	if !equality.Semantic.DeepEqual(vmRestore.Status, vmRestoreCpy.Status) {
+		return ctrl.doUpdateStatus(vmRestore, vmRestoreCpy)
+	}
+
 	controller.RemoveFinalizer(vmRestoreCpy, vmRestoreFinalizer)
-	return ctrl.doUpdate(vmRestore, vmRestoreCpy)
+	_, err := ctrl.Client.VirtualMachineRestore(vmRestore.Namespace).Update(context.Background(), vmRestoreCpy, metav1.UpdateOptions{})
```

Comment:
@mhenriks why not using patch here?

Reply:
+1

Reply:
Was thinking about it but figured that the likelihood of a conflict is low since mutating the result of previous `updateStatus()` call a couple lines before. But may as well do patch to be consistent with how snapshot finalizers are applied